lib/WPA/FlowSensitive.cpp

```diff
@@ -612,8 +612,32 @@ bool FlowSensitive::isStrongUpdate(const SVFGNode* node, NodeID& singleton)
 bool FlowSensitive::updateCallGraph(const CallSiteToFunPtrMap& callsites)
 {
     double start = stat->getClk();
+    CallEdgeMap potentialNewEdges;
+    onTheFlyCallGraphSolve(callsites, potentialNewEdges);
+
+    // Bound the new edges by the Andersen's call graph.
+    // TODO: we want this to be an assertion.
+    const CallEdgeMap &andersCallEdgeMap = ander->getIndCallMap();
     CallEdgeMap newEdges;
-    onTheFlyCallGraphSolve(callsites, newEdges);
```

Comment:
Can we still invoke this function and then filter out the resulting new edges? This will be consistent with other pointer analyses implementations.

Reply:
The filtering can be a separate function in order to be more flexible to change to an assertion for soundness later.

Reply:
Hmm I don't fully understand. Currently it passes in potentialNewEdges into this function, then builds newEdges from that. Do you mean to pass in newEdges then remove elements from it?

Reply:
Yes, I can make it its own function, what kind of API do you envision?

Reply:
> Hmm I don't fully understand. Currently it passes in potentialNewEdges into this function, then builds newEdges from that. Do you mean to pass in newEdges then remove elements from it?

Yes

Reply:
> Yes, I can make it its own function, what kind of API do you envision?

If the implementation is short, we can just put it in this function.

Reply:
Hmm, can't be in `onTheFlyCallGraphSolve` because it has no notion of auxiliary analysis.

Reply:
No, I mean inside `FlowSensitive::updateCallGraph`

Reply:
Oh okay, it is 30 lines now, I'm just testing now before pushing.